third_party/txt/src/txt/paragraph.cc

```diff
@@ -571,9 +589,15 @@ void Paragraph::Layout(double width, bool force) {
           line_range.end_excluding_whitespace < line_range.end &&
           bidi_run.start() <= line_range.end &&
           bidi_run.end() > line_end_index) {
+        // Use the most recent TextDirection instead of the defaulted LTR value
```

Comment:
`ParagraphStyle` has a field that sets the base `TextDirection` for the paragraph.  What happens in this case if you set that field to RTL?